mock-plugin/src/test/java/io/mosip/signup/plugin/mock/service/MockIdentityVerifierPluginImplTest.java

```diff
@@ -0,0 +1,119 @@
+        frameDetail.setFrame("frame");
+        frameDetail.setOrder(0);
+        frameDetails.add(frameDetail);
+        identityVerificationDto.setFrames(frameDetails);
+        MockUserStory mockUserStory = new MockUserStory();
+        List<MockScene> mockScenes = new ArrayList<>();
+        MockScene mockScene = new MockScene();
+        mockScene.setFrameNumber(0);
+        mockScene.setStepCode("START");
+        mockScenes.add(mockScene);
+        mockUserStory.setScenes(mockScenes);
+        Mockito.when(restTemplate.getForObject(Mockito.anyString(), Mockito.eq(MockUserStory.class))).thenReturn(mockUserStory);
+        mockIdentityVerifierPlugin.verify(transactionId, identityVerificationDto);
+
+        Mockito.verify(restTemplate, Mockito.times(1)).getForObject(Mockito.anyString(), Mockito.eq(MockUserStory.class));
+    }
+
+
+    @Test
+    public void getVerifiedResult_withValidTransactionId_thenPass() throws IdentityVerifierException, JsonProcessingException {
+
+        ObjectMapper objectMapper1 = new ObjectMapper();
+        ReflectionTestUtils.setField(mockIdentityVerifierPlugin, "objectMapper",objectMapper1);
+        String transactionId = "transactionId123";
+        MockUserStory mockUserStory = new MockUserStory();
+        VerifiedResult verifiedResult = new VerifiedResult();
+        Map<String, VerificationDetail> verificationDetailMap = new HashMap<>();
+        VerificationDetail verificationDetail = new VerificationDetail();
+        verificationDetail.setTrust_framework("trust_framework");
+        verificationDetail.setVerification_process("verification_process");
+        verificationDetailMap.put("claim", verificationDetail);
+        verifiedResult.setVerifiedClaims(verificationDetailMap);
+        verifiedResult.setStatus(VerificationStatus.COMPLETED);
+
+        JsonNode jsonNode =objectMapper1.readTree(objectMapper1.writeValueAsString(verifiedResult));
+
+        mockUserStory.setVerifiedResult(jsonNode);
+        List<MockScene> mockScenes = new ArrayList<>();
+        MockScene mockScene = new MockScene();
+        mockScene.setFrameNumber(0);
+        mockScene.setStepCode("START");
+        mockScenes.add(mockScene);
+
+        mockUserStory.setScenes(mockScenes);
+
+        Mockito.when(restTemplate.getForObject(Mockito.anyString(), Mockito.eq(MockUserStory.class))).thenReturn(mockUserStory);
+        VerifiedResult actualVerifiedResult = mockIdentityVerifierPlugin.getVerifiedResult(transactionId);
+
+        Assert.assertEquals(verifiedResult.getStatus(), actualVerifiedResult.getStatus());
+        Assert.assertEquals(verifiedResult.getVerifiedClaims(), actualVerifiedResult.getVerifiedClaims());
+        Mockito.verify(restTemplate, Mockito.times(1)).getForObject(Mockito.anyString(), Mockito.eq(MockUserStory.class));
+    }
+
+
+    @Test
+    public void getVerifiedResult_withValidTransactionId_thenFail() throws IdentityVerifierException {
+
+        ObjectMapper objectMapper1 = new ObjectMapper();
```

Comment:
why is it named objectMapper1? if its required for more than 1 test case, lets define it as class level variable.

Reply:
changed